mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py

```diff
@@ -212,33 +290,97 @@ def verbose_name(self):
     @api.model
     def create(self, vals):
         vals.update({
-            'ref': self.env['ir.sequence'].get('mgmtsystem.nonconformity')
+            'ref': self.env['ir.sequence'].next_by_code(
+                'mgmtsystem.nonconformity')
         })
+
         return super(MgmtsystemNonconformity, self).create(vals)
 
     @api.multi
     def message_auto_subscribe(self, updated_fields, values=None):
-        """Add the responsible, manager and OpenChatter follow list."""
+        """
+        Add the responsible, manager and author to the OpenChatter follow list
+        """
         self.ensure_one()
-        user_ids = [
-            self.responsible_user_id.id,
-            self.manager_user_id.id,
-            self.author_user_id.id,
-        ]
-        self.message_subscribe_users(user_ids=user_ids, subtype_ids=None)
+
+        # user_ids = [
+        #    self.responsible_user_id._uid,
+        #    self.manager_user_id._uid,
+        #    self.author_user_id._uid,
+        # ]
+        # self.message_subscribe_users(user_ids=user_ids)
+
         return super(MgmtsystemNonconformity, self).message_auto_subscribe(
             updated_fields=updated_fields,
             values=values
         )
 
     @api.multi
-    def wkf_analysis(self):
-        """Change state from draft to analysis"""
-        return self.write({
+    def write(self, vals):
+        """Update user data."""
+        if vals.get('state') or vals.get('state_id'):
+            if vals.get('state') == "analysis" or vals.get(
+                    'state_id') == self.env.ref(
+                    "mgmtsystem_nonconformity.state_analysis").id:
+                vals.update(self.do_analysis())
+            if vals.get('state') == "pending" or vals.get(
+                    'state_id') == self.env.ref(
+                    "mgmtsystem_nonconformity.state_pending").id:
+                vals.update(self.do_review())
+            if vals.get('state') == "open" or vals.get(
+                    'state_id') == self.env.ref(
+                    "mgmtsystem_nonconformity.state_open").id:
+                vals.update(self.do_open())
+            if vals.get('state') == "done" or vals.get(
+                    'state_id') == self.env.ref(
+                    "mgmtsystem_nonconformity.state_done").id:
+                vals.update(self.do_close())
+            if vals.get('state') == "cancel" or vals.get(
+                    'state_id') == self.env.ref(
+                    "mgmtsystem_nonconformity.state_cancel").id:
+                vals.update(self.do_cancel())
+            if vals.get('state') == "draft" or vals.get(
+                    'state_id') == self.env.ref(
+                    "mgmtsystem_nonconformity.state_draft").id:
+                vals.update(self.case_reset())
```

Comment:
If you still prefer to use Stages (and that is the proper solution for Kanbans) you will still need to keep a computed `stage` field, so that the logic works properly.

This means that you have a `stage_id` on conformities, and a Stages tables.
And then you must map each Stage to a State, and have a `related` field on Nonconformities that automatically computes it's `state`  from the document's `stage_id`.
I implemented this for Projects, here: https://github.com/OCA/project/tree/8.0/project_stage_state